core/mount/mount_linux_test.go

```diff
@@ -243,3 +352,46 @@ func setupMounts(t *testing.T) (target string, mounts []Mount) {
 
 	return target, mounts
 }
+
+func supportsIDMap(path string) bool {
```

Comment:
Only glancing over, but wondering if there's expected and _unexpected_ errors that could happen in this helper that we may be hiding?

Mostly considering if this helper should return an error (instead of a bool), and handle _expected_ errors, and return unexpected ones, so that they can be logged in the caller (possibly even make the test fail).

Skipping a test may be easy to overlook ("skipping test because doesn't support idmapped mounts") but could hide actual issues that need looking into?

Reply:
I think this function is copied from https://github.com/containerd/containerd/blob/9b89fb989d91252459cf4c78ac23e2d5b07d2263/integration/pod_userns_linux_test.go#L51

I think we can make exported function in the follow-up to unify them.

Reply:
Indeed, this was copied. Does it feel best to tinker with all copies of it in this PR or I could also look at doing that separately with a new issue. I see your point @thaJeztah  and it seems easy enough to tighten up either way.

Reply:
ping @thaJeztah let's handle this in the follow-up. WDYT?